doc/developer-guide/api/functions/TSTypes.en.rst

```diff
@@ -78,6 +78,34 @@ more widely. Those are described on this page.
 
    A 64 bit time value, measured in nanoseconds.
 
+.. type:: TSHttpConnectPluginOptions
```

Comment:
This needs a leading value that indicates what data is valid in it for backwards compatibility. This could be a size, or a version, or both. This must always be first (similar to `sa_family` in `sockaddr`). Alternatively this length could be passed as an argument to `TSHttpConnectPlugin`.

Reply:
A way to mark values as "default" or "not set" is needed.

Reply:
I _think_ I have captured what you've asked for here in the last commit I just pushed. Please take a look.

Reply:
@SolidWallOfCode Please take a look at the `TSHttpConnectOptionsGet(TSConnectType connect_type)` API function and its uses.